File: src/main/java/net/sjrx/intellij/plugins/systemdunitfiles/semanticdata/optionvalues/NullOptionValue.java
package net.sjrx.intellij.plugins.systemdunitfiles.semanticdata.optionvalues;

import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.NotNull;

import java.util.Collections;
import java.util.Set;

/**
 * This validator has no auto completion and marks everything as valid.
 */
public class NullOptionValue implements OptionValueInformation {
  
  @Override
  public Set<String> getAutoCompleteOptions(@NotNull Project project) {
    return Collections.emptySet();
  }
  
  @Override
  public String getErrorMessage(String value) {
    return null;
  }
  
  @Override
  public String getValidatorName() {
    return "NULL";
  }
}

File: src/main/kotlin/net/sjrx/intellij/plugins/systemdunitfiles/semanticdata/optionvalues/NullOptionValue.kt
package net.sjrx.intellij.plugins.systemdunitfiles.semanticdata.optionvalues

import com.intellij.openapi.project.Project
import net.sjrx.intellij.plugins.systemdunitfiles.semanticdata.Validator

/**
 * This validator has no auto completion and marks everything as valid.
 */
class NullOptionValue : OptionValueInformation {
  override fun getAutoCompleteOptions(project: Project): Set<String> {
    return emptySet()
  }

  override fun getErrorMessage(value: String): String? {
    return null
  }

  override val validatorName: String
    get() = "NULL"

  companion object {
    const val VALIDATOR_NAME = "NULL"
    val INSTANCE = NullOptionValue()
    val VALIDATOR_INSTANCE = Validator(VALIDATOR_NAME, "0")
    val validators = mapOf(VALIDATOR_INSTANCE to INSTANCE, Validator("config_parse_warn_compat", "*") to INSTANCE)
  }
}
